lib/App.js

```diff
@@ -176,34 +176,74 @@ App.prototype.loadViews = function() {};
 
 App.prototype.loadStyles = function() {};
 
+// This function is defined by requiring 'derby/parsing'
+App.prototype.addViews = null;
+
 App.prototype.component = function(viewName, constructor) {
   if (typeof viewName === 'function') {
     constructor = viewName;
     viewName = null;
   }
-  // DEPRECATED: constructor.prototype.view and constructor.prototype.name
-  var viewFilename = constructor.view || constructor.prototype.view;
-  var constructorIs = constructor.is || constructor.prototype.name;
+
+  var viewArg = constructor.view;
+  var viewIs, viewFilename, viewSource, viewDependencies;
+  // Always using an object for the static `view` property is preferred
+  if (viewArg && typeof viewArg === 'object') {
+    viewIs = viewArg.is;
+    viewFilename = viewArg.file;
+    viewSource = viewArg.source;
+    viewDependencies = viewArg.dependencies;
+  } else {
+    // Ignore other properties when `view` is an object. It is possible that
+    // properties could be inherited from a parent component when extending it.
+    //
+    // DEPRECATED: constructor.prototype.name and constructor.prototype.view
+    // use the equivalent static properties instead
+    viewIs = constructor.is || constructor.prototype.name;
+    viewFilename = constructor.view || constructor.prototype.view;
+  }
+
+  if (viewFilename && viewSource) {
+    throw new Error('Component may not specify both a view file and source');
+  }
+
+  if (viewDependencies) {
+    for (var i = 0; i < viewDependencies.length; i++) {
+      var dependency = viewDependencies[i];
+      if (Array.isArray(dependency)) {
+        this.component(dependency[0], dependency[1]);
+      } else {
+        this.component(dependency);
+      }
+    }
+  }
 
   // Inherit from Component
   components.extendComponent(constructor);
 
+  var view;
   if (viewFilename) {
-    viewName = viewName || constructorIs || path.basename(viewFilename, '.html');
+    viewName = viewName || viewIs || path.basename(viewFilename, '.html');
     this.loadViews(viewFilename, viewName);
+    view = this.views.find(viewName);
 
-  } else if (!viewName) {
-    if (constructorIs) {
-      viewName = constructorIs;
-      var view = this.views.register(viewName);
-      view.template = templates.emptyTemplate;
-    } else {
-      throw new Error('No view name specified for component');
-    }
+  } else if (viewSource) {
+    viewName = viewName || viewIs;
+    this.addViews(viewSource, viewName);
```

Comment:
This would throw an error "this.addViews is not a function" if derby-parsing isn't registered.

It'd be nice to have a friendlier error message. How about, instead of initializing `App.prototype.addViews` to null above, instead initialize it to a function that throws a more actionable error - something like "derby-parsing not available. Component view.viewSource should not be used in production. Instead, use a filename with view.file."

Reply:
Good idea.